node/test/util/stash_util.js

```diff
@@ -820,4 +822,50 @@ x=E:Cm-1 s=Sa:s;Bm=m;Os Cs foo=bar!Bs=s!W foo=bar`
             }));
         });
     });
+    describe("shadowCommit", function () {
```

Comment:
Extend the existing `"shadow"` test cases to support timestamp. You don't need to validate that two commits compare equal in the test driver, just that the one commit generated has the expected timestamp, if provided (you should validate this for sanity on the CL that the SHAs are the same, in case there's something weird libgit2 is doing).